### ISSUE TITLE
Make RedisRecorderActor reconnect

### ISSUE THREAD

**Issue by ritzalam** [bigbluebutton/bigbluebutton#9225]
**Describe the bug**
When the recording actor is disconnected from redis, it doesn't auto-reconnect thus loosing events.
Happens when KMS takes all memory and redis cannot fork to persists data.

**To Reproduce**
Make another process gobble up memory so Redis fails to fork. 
Or, have a way to kill recording actor redis connection.

**Expected behavior**
Make connection reconnect.

**Actual behavior**
Connection should re-establish and be able to continuously save events.

**Comment by frsteinb** [comment]
Could you please help to get a better understanding... Is this issue related to this Exceptions shown by `bbb-conf --debug`:

`/var/log/bigbluebutton/bbb-web.2020-04-24.log:2020-04-24T23:59:13.304+02:00 INFO  o.b.api2.util.RecMetaXmlHelper - Exception while saving /var/bigbluebutton/published/presentation/f2feb667adaeb585b2de373b07f0d0624845f791-1587732292524/metadata.xml
`

These are the exception details:

`2020-04-24T23:59:13.304+02:00 INFO  o.b.api2.util.RecMetaXmlHelper - Exception details: {}
java.lang.IllegalArgumentException: For input string: ""
        at org.bigbluebutton.api2.util.RecMetaXmlHelper.getRecordingMetadata(RecMetaXmlHelper.scala:92)
        at org.bigbluebutton.api.util.RecordingMetadataReaderHelper.getRecordingMetadata(RecordingMetadataReaderHelper.java:39)
        at org.bigbluebutton.api.RecordingService.getRecordingMetadata(RecordingService.java:193)
        at org.bigbluebutton.api.RecordingService.getRecordingsMetadata(RecordingService.java:162)
        at org.bigbluebutton.api.RecordingService.getRecordings2x(RecordingService.java:186)
        at org.bigbluebutton.api.MeetingService.getRecordings2x(MeetingService.java:450)
        at org.bigbluebutton.api.MeetingService$getRecordings2x$1.call(Unknown Source)
        at org.bigbluebutton.web.controllers.ApiController$_closure18.doCall(ApiController.groovy:1744)
        at org.bigbluebutton.web.controllers.ApiController$_closure18.doCall(ApiController.groovy)
        at sun.reflect.GeneratedMethodAccessor88.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at groovy.lang.Closure.call(Closure.java:418)
        at groovy.lang.Closure.call(Closure.java:412)
        at org.bigbluebutton.web.controllers.ApiController.getRecordingsHandler(ApiController.groovy)
        at org.grails.core.DefaultGrailsControllerClass$MethodHandleInvoker.invoke(DefaultGrailsControllerClass.java:223)
        at org.grails.core.DefaultGrailsControllerClass.invoke(DefaultGrailsControllerClass.java:188)
        at org.grails.web.mapping.mvc.UrlMappingsInfoHandlerAdapter.handle(UrlMappingsInfoHandlerAdapter.groovy:90)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.boot.actuate.trace.WebRequestTraceFilter.doFilterInternal(WebRequestTraceFilter.java:111)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:77)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:96)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.boot.actuate.autoconfigure.MetricsFilter.doFilterInternal(MetricsFilter.java:103)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
`

**Comment by ritzalam** [comment]
No. THis exception seems to be a result of changing the metadata.xml of a recording. It fails to save the metadata.xml after the change.

**Comment by ritzalam** [comment]
This is the log that appears in `bbb-apps-akka.log`

```
2020-02-12T20:57:17.012-05:00 ERROR akka.actor.OneForOneStrategy - MISCONF Redis is configured to save RDB snapshots, but is currently not able to persist on disk.
```

Here is the entries on redis.log

![redis-cannot-save-events](https://user-images.githubusercontent.com/184672/80278844-88cd6f00-86c7-11ea-89ba-65b7afaa890a.PNG)

**Comment by frsteinb** [comment]
Hm. In my case, these messages cannot be found in my logfiles. Neither redis, nor apps-akka.

( I checked the time range and grep'ed like this: `egrep -r '(OneForOneStrategy|temporarily unavailable)' /var/log/ `)

However, I found this message and followed the advice:

`23585:M 20 Apr 17:27:50.094 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
`
